frontend/src/component/UIFactory.tsx

```diff
@@ -130,6 +138,17 @@ export class UIFactory {
     );
   }
 
+  public createPublicListingToggle(props: any): ReactElement {
+    let ToggleLabel = (
+      <span>
+        Share on <span>public feed</span>
+      </span>
+    );
+    return (
+      <this.ToggledPublicListingToggle toggleLabel={ToggleLabel} {...props} />
+    );
+  }
```

Comment:
This should be a separate component.

Reply:
The toggle label you mean right?

Reply:
@Coteh The toggle and the label together.

Reply:
I believe I already did this, it's called `PreferenceToggle`. It encapsulates a `Toggle` and a label.

Reply:
@Coteh Sorry, what I mean is to create `PublicListingToggle`

Reply:
Right now a `PublicListingToggle` is just a toggle with a label that triggers an `onClick` function we pass down (from the `Home` component). I don't see any behaviour for `PublicListingToggle` that makes it any different from any other `PreferenceToggle`. Therefore, I think we should either keep it as `PreferenceToggle`, or rename `PreferenceToggle` to `PublicListingToggle` since we don't currently use `PreferenceToggle` for anything else at the moment.

Reply:
@Coteh Let's rename `PreferenceToggle` to `PublicListingToggle` and hard code the label inside.

Reply:
Done